Staging - changes for the staging node

adean20csc opened the pull request:
The initial_stage.yml will create the directory tree on the usb drive and download the latest tagged version of the public git repos

The stage_usb.yml will be run from the usb drive Kragle and will go out and download all other resources to the usb drive

```
[root@stage-node ~]# ansible-playbook initial_stage.yml

PLAY [Download the Kragle source repo] ****************************************

TASK: [Create the staging directories on the usb hard drive] ******************
ok: [localhost] => (item=/mnt/sdb/autodeploy/projects)
ok: [localhost] => (item=/mnt/sdb/autodeploy/resources)

TASK: [Determine the latest tagged versions of public source Git repos] *******
ok: [localhost] => (item={'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'})
ok: [localhost] => (item={'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'})
ok: [localhost] => (item={'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'})

TASK: [Clone the public project repos from Git] *******************************
changed: [localhost] => (item={u'content_length': u'635', u'status': 200, u'x_github_request_id': u'470E7AF9:AB2E:13FCD0FA:56291784', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/kragle/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'7f48e2f7761567e923121f17538d7a6d', u'access_control_allow_origin': u'*', u'last_modified': u'Thu, 27 Aug 2015 12:41:21 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Thu, 22 Oct 2015 17:06:13 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445536970', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/kragle/commits/32cf7e96f841c7f8a7cb5d0a9ac9e9761796263a', u'sha': u'32cf7e96f841c7f8a7cb5d0a9ac9e9761796263a'}, u'zipball_url': u'https://api.github.com/repos/csc/kragle/zipball/v0.1.0', u'tarball_url': u'https://api.github.com/repos/csc/kragle/tarball/v0.1.0', u'name': u'v0.1.0'}, {u'commit': {u'url': u'https://api.github.com/repos/csc/kragle/commits/e36116c507a169c5a7efb2b87e51e1b47c096b57', u'sha': u'e36116c507a169c5a7efb2b87e51e1b47c096b57'}, u'zipball_url': u'https://api.github.com/repos/csc/kragle/zipball/v0.0.1', u'tarball_url': u'https://api.github.com/repos/csc/kragle/tarball/v0.0.1', u'name': u'v0.0.1'}], u'etag': u'W/"825883ced92037439608f59c8e752e79"', 'item': {'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'53'})
changed: [localhost] => (item={u'content_length': u'635', u'status': 200, u'x_github_request_id': u'470E7AF9:AB2F:1412EEAF:56291786', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/slimer/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'318e55760cf7cdb40e61175a4d36cd32', u'access_control_allow_origin': u'*', u'last_modified': u'Mon, 01 Jun 2015 14:37:54 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Thu, 22 Oct 2015 17:06:15 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445536970', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/slimer/commits/e49ad41c8de4185b6134988f59a32f15ab602cb2', u'sha': u'e49ad41c8de4185b6134988f59a32f15ab602cb2'}, u'zipball_url': u'https://api.github.com/repos/csc/slimer/zipball/v0.1.2', u'tarball_url': u'https://api.github.com/repos/csc/slimer/tarball/v0.1.2', u'name': u'v0.1.2'}, {u'commit': {u'url': u'https://api.github.com/repos/csc/slimer/commits/800ddb77080610c267ec776857559379c54c0afe', u'sha': u'800ddb77080610c267ec776857559379c54c0afe'}, u'zipball_url': u'https://api.github.com/repos/csc/slimer/zipball/v0.1.1', u'tarball_url': u'https://api.github.com/repos/csc/slimer/tarball/v0.1.1', u'name': u'v0.1.1'}], u'etag': u'W/"bfb97f0b6dfa0adee9a71e692145fdd9"', 'item': {'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'52'})
changed: [localhost] => (item={u'content_length': u'345', u'status': 200, u'x_github_request_id': u'470E7AF9:AB2E:13FCD4EC:56291788', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/ansible-scaleio/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'318e55760cf7cdb40e61175a4d36cd32', u'access_control_allow_origin': u'*', u'last_modified': u'Fri, 07 Aug 2015 00:35:06 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Thu, 22 Oct 2015 17:06:17 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445536970', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/ansible-scaleio/commits/15aca7e868546f09bf219828662b93887041fbaf', u'sha': u'15aca7e868546f09bf219828662b93887041fbaf'}, u'zipball_url': u'https://api.github.com/repos/csc/ansible-scaleio/zipball/v0.1.1', u'tarball_url': u'https://api.github.com/repos/csc/ansible-scaleio/tarball/v0.1.1', u'name': u'v0.1.1'}], u'etag': u'W/"de1dbf582f2129ceee747a36e29e9089"', 'item': {'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'51'})

PLAY RECAP ********************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0
```

mtnbikenc commented on ansible/initial_stage.yml:
```diff
@@ -0,0 +1,46 @@
+  hosts: localhost
+  gather_facts: false
+
+  vars:
```
@adean20csc Why are these vars here as well as in group_vars/all.yml? Lines 10-18

adean20csc commented:
```

[root@stage-node ~]# ansible-playbook /opt/autodeploy/projects/kragle/ansible/stage_usb.yml

PLAY [Prepare the staging node] ***********************************************

GATHERING FACTS ***************************************************************
ok: [localhost]

TASK: [install support packages] **********************************************
ok: [localhost] => (item=bridge-utils,dhcp,docker,ipmitool,iptables-services,ntp,python-ipaddr,python-passlib,sshpass,vim,httpd)

TASK: [install support packages with pip] *************************************
ok: [localhost] => (item=docker-py)
ok: [localhost] => (item=selenium)
ok: [localhost] => (item=pip2pi)

TASK: [Create the staging directories on the usb hard drive] ******************
changed: [localhost] => (item=/mnt/sdb/autodeploy/projects)
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources)

TASK: [Create the resource directories in the staging directories] ************
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources/docker)
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources/ISO)
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources/packages)
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources/rpms)
changed: [localhost] => (item=/mnt/sdb/autodeploy/resources/scaleio)

TASK: [Download the private project repo from Git] ****************************
ok: [localhost]

TASK: [Download the Hanlon microkernel image] *********************************
changed: [localhost]

TASK: [Find the RHEL DVD ISO file url] ****************************************
ok: [localhost]

TASK: [Download the RHEL DVD ISO file] ****************************************
changed: [localhost]

TASK: [Find the Red Hat KVM Guest Image file url] *****************************
ok: [localhost]

TASK: [Download the KVM Guest Image file] *************************************
changed: [localhost]

TASK: [Download the Red Hat RPMs] *********************************************
changed: [localhost]

TASK: [Create the local offline repository] ***********************************
changed: [localhost]

TASK: [Mirror docker-py and its dependencies] *********************************
changed: [localhost]

TASK: [Create the PyPi-compatible "simple" package] ***************************
changed: [localhost]

TASK: [Start and enable the docker service] ***********************************
ok: [localhost]

TASK: [Pull the project docker images from the docker registry] ***************
ok: [localhost] => (item={'image': 'mongo', 'name': 'hanlon-mongo'})
ok: [localhost] => (item={'image': 'cscdock/hanlon', 'name': 'hanlon-server'})
ok: [localhost] => (item={'image': 'cscdock/atftpd', 'name': 'hanlon-atftpd'})

TASK: [Save container image to usb drive] *************************************
changed: [localhost] => (item={'image': 'cscdock/hanlon', 'tar_file': 'cscdock_hanlon.tar'})
changed: [localhost] => (item={'image': 'cscdock/atftpd', 'tar_file': 'cscdock_atftpd.tar'})
changed: [localhost] => (item={'image': 'mongo', 'tar_file': 'mongo.tar'})

TASK: [Download ScaleIO files for ansible-scaleio project] ********************
changed: [localhost]

PLAY RECAP ********************************************************************
localhost                  : ok=19   changed=11   unreachable=0    failed=0
```

mtnbikenc commented:
Code review :+1:

jcpowermac commented:
Code Review :+1:
